src/loader.py

```diff
@@ -100,26 +99,49 @@ async def summarize_document(doc, client):
 }
 ```
 """.strip()
+    
+    content = {"content": doc.text, **doc.metadata}
 
-    max_retries = 5
-    attempt = 0
-    while attempt < max_retries:
-        try:
-            chat_completion = client.chat.completions.create(
-                messages=[
-                    {"role": "system", "content": PROMPT},
-                    {"role": "user", "content": json.dumps(doc)},
-                ],
-                model="llama-3.1-70b-versatile",
-                response_format={"type": "json_object"},
-                temperature=0,
-            )
-            break
-        except Exception as e:
-            print("Error status {}".format(e.status_code))
-            attempt += 1
+    client = ollama.AsyncClient()
+    chat_completion = await client.chat(
+        messages = [
+            {
+                "role": "system",
+                "content": PROMPT},
+            {
+                "role": "user",
+                "content": json.dumps(content),
+            },
+        ],
+        model = "llama3.2",
+        format = "json",
```

Comment:
I have not run this yet, but I'm not sure llama3.2 has json mode support

Reply:
I think it might? Not certain
https://github.com/ollama/ollama/blob/main/docs/api.md#request-json-mode